### PR TITLE
Handling USR ( user signup restrict ) error code

### PR DESCRIPTION
**Pull request by mohamed-siddhiq**
## What

- Handling the USR (User Signup Restrict) error code that arises when signup restrictions are enabled. 

## Why

-

## How

-

## Database Migrations

- 

## Env Config

- 

## Relevant Docs

-

## Related Issues or PRs

-

## Dependencies Versions

-

## Notes on Testing

-

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

**Review by johnyrahul — APPROVED** [review]
LGTM

File: backend/account/authentication_controller.py [hunk 1]
```diff
@@ -162,9 +162,14 @@ def user_organizations(self, request: Request) -> Any:
             if ex.code == AuthorizationErrorCode.USF:  # type: ignore
                 response = Response(
                     status=status.HTTP_412_PRECONDITION_FAILED,
-                    data={"domain": ex.data.get("domain")},  # type: ignore
+                    data={"domain": ex.data.get("domain"), "code": ex.code},  # type: ignore
                 )
-                return response
+            elif ex.code == AuthorizationErrorCode.USR:  # type: ignore
```
Comment (muhammad-ali-e) [inline backend/account/authentication_controller.py:167]: seems duplicating the if block here.. can we avoid the duplication here ?

File: frontend/src/hooks/useSessionValid.js [hunk 2]
```diff
@@ -114,7 +114,8 @@ function useSessionValid() {
 
       if (err.request?.status === 412) {
         const domainName = JSON.parse(err.request.response).domain;
-        window.location.href = `/error?code=USF&domain=${domainName}`;
+        const code = JSON.parse(err.request.response).code;
```
Comment (muhammad-ali-e) [inline frontend/src/hooks/useSessionValid.js:117]: NIT:    better to do like this
``` js
    const response = JSON.parse(err.request.response);
    const domainName = response.domain;
    const code = response.code;
```

**Review by athul-rs — APPROVED** [review]

**Comment by athul-rs** [comment]
@mohamed-siddhiq Please resolve the comment @muhammad-ali-e left, before merging.

File: frontend/src/hooks/useSessionValid.js (hunk 2 above)
Comment (mohamed-siddhiq) [reply frontend/src/hooks/useSessionValid.js]: @muhammad-ali-e  addressed your comments.

**Review by nehabagdia — APPROVED** [review]

**Pull request merged**